src/plugins/platforms/eglfs/qeglfscontext.cpp

```diff
@@ -48,15 +48,212 @@
+
+        glFuncs.glUseProgram(m_program);
+        glActiveTexture(m_activeTextureUnit);
+        glBindTexture(GL_TEXTURE_2D, m_texture);
+        glFuncs.glBindFramebuffer(GL_FRAMEBUFFER, m_fbo);
+        glViewport(m_viewport[0], m_viewport[1], m_viewport[2], m_viewport[3]);
+        glDepthMask(m_depthWriteMask);
+        glColorMask(m_colorWriteMask[0], m_colorWriteMask[1], m_colorWriteMask[2], m_colorWriteMask[3]);
+        if (m_blend)
+            glEnable(GL_BLEND);
+        if (m_depth)
+            glEnable(GL_DEPTH_TEST);
+        if (m_cull)
+            glEnable(GL_CULL_FACE);
+        if (m_scissor)
+            glEnable(GL_SCISSOR_TEST);
+        for (int i = 0; i < STATE_GUARD_VERTEX_ATTRIB_COUNT; ++i) {
+            if (m_vertexAttribs[i].enabled)
+                glFuncs.glEnableVertexAttribArray(i);
+            GLuint prevBuf;
+            glGetIntegerv(GL_ARRAY_BUFFER_BINDING, (GLint *) &prevBuf);
+            glFuncs.glBindBuffer(GL_ARRAY_BUFFER, m_vertexAttribs[i].arrayBuffer);
+            glFuncs.glVertexAttribPointer(i, m_vertexAttribs[i].size, m_vertexAttribs[i].type,
+                                          m_vertexAttribs[i].normalized, m_vertexAttribs[i].stride,
+                                          m_vertexAttribs[i].pointer);
+            glFuncs.glBindBuffer(GL_ARRAY_BUFFER, prevBuf);
+        }
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, m_minFilter);
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, m_magFilter);
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, m_wrapS);
+        glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, m_wrapT);
+    }
+
+private:
+    GLuint m_program;
+    GLenum m_activeTextureUnit;
+    GLuint m_texture;
+    GLuint m_fbo;
+    GLint m_depthWriteMask;
+    GLint m_colorWriteMask[4];
+    GLboolean m_blend;
+    GLboolean m_depth;
+    GLboolean m_cull;
+    GLboolean m_scissor;
+    GLint m_viewport[4];
+    struct VertexAttrib {
+        bool enabled;
+        GLuint arrayBuffer;
+        GLint size;
+        GLint stride;
+        GLenum type;
+        bool normalized;
+        void *pointer;
+    } m_vertexAttribs[STATE_GUARD_VERTEX_ATTRIB_COUNT];
+    GLenum m_minFilter;
+    GLenum m_magFilter;
+    GLenum m_wrapS;
+    GLenum m_wrapT;
+};
+
 QEglFSContext::QEglFSContext(const QSurfaceFormat &format, QPlatformOpenGLContext *share,
                              EGLDisplay display, EGLenum eglApi)
     : QEGLPlatformContext(QEglFSHooks::hooks()->surfaceFormatFor(format), share, display,
                           QEglFSIntegration::chooseConfig(display, QEglFSHooks::hooks()->surfaceFormatFor(format)), eglApi),
-      m_swapIntervalSet(false)
+      m_swapIntervalSet(false),
+      m_useNativeDefaultFbo(false),
+      m_fbo(0),
+      m_blitter(0)
+{
+    m_scale = qgetenv("QT_QPA_EGLFS_SCALE").toInt();
+}
+
+QEglFSContext::~QEglFSContext()
 {
+    delete m_blitter;
```

Comment:
delete m_fbo as well?

Or possibly use QScopedPointer for m_blitter and m_fbo.